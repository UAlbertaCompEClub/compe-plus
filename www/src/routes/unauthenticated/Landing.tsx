import { useAuth0 } from '@auth0/auth0-react';
import { Box, Button, Grid, Link, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import React, { FC } from 'react';

import Industry from '../../assets/industry.svg';
import BlackLogo from '../../assets/logo_black.svg';
import LightGreenLogo from '../../assets/logo_light_green.svg';
import MockInterview from '../../assets/mock_interview.svg';
import PairProgramming from '../../assets/pair_programming.svg';
import Practice from '../../assets/practice.svg';
import ResumeReview from '../../assets/resume_review.svg';
import Support from '../../assets/support.svg';
import Teamwork from '../../assets/teamwork.svg';
import Wave1 from '../../assets/wave_1.svg';
import Wave2 from '../../assets/wave_2.svg';
import Wave3 from '../../assets/wave_3.svg';
import Wave4 from '../../assets/wave_4.svg';
import { Fade } from '../../components/Fade';
import MailChimpForm from '../../components/MailChimpForm';
import useGlobalStyles from '../../styles/style';

const Landing: FC = () => {
    const classes = useStyles();

    return (
        <Grid container direction='row' justify='center' style={{ overflowX: 'hidden' }}>
            <Intro />
            <img src={Wave1} className={classes.wave} />
            <Services />
            <Info />
            <CallToAction />
            <Values />
            <img src={Wave4} className={classes.wave} />
            <Footer />
        </Grid>
    );
};

export const Intro: FC = () => {
    const classes = useStyles();
    const { isAuthenticated } = useAuth0();
    const global = useGlobalStyles();

    return (
        <Grid container item className={classes.wave_pattern} justify='center' style={{ minHeight: '50vh', paddingTop: '10vh' }} id='intro'>
            <Grid container item justify='center' spacing={5}>
                <Grid container item xs={4} alignItems='center' justify='center'>
                    <Grid item>
                        <Box maxWidth='sm'>
                            <img src={BlackLogo} className={classes.logo_1} />
                        </Box>
                        <Typography variant='h1'>CompE+</Typography>
                        <Typography variant='h3'>Resume Review, Mock Interviews, Community</Typography>
                    </Grid>
                </Grid>

                <Grid container item xs={5} alignItems='center' justify='center'>
                    <img src={PairProgramming} className={classes.svg} />
                </Grid>
            </Grid>
            <Grid container item justify='center' alignItems='center' style={{ paddingTop: '5vh' }} spacing={5}>
                <Grid container item justify='center' alignItems='center' spacing={4}>
                    {!isAuthenticated && (
                        <Grid item>
                            <Button size='large' variant='contained' className={global.main_button} onClick={() => window.location.replace('/#call-to-action')}>
                                Get Started!
                            </Button>
                        </Grid>
                    )}
                    <Grid item>
                        <Button size='large' variant='outlined' className={global.secondary_button} onClick={() => window.location.replace('/#services')}>
                            Learn More
                        </Button>
                    </Grid>
                </Grid>
                <Grid container item justify='center' alignItems='center'>
                    <Grid item>
                        <Typography variant='h3' style={{ fontSize: 28 }}>
                            Get the help you need to land your dream internship!
                        </Typography>
                    </Grid>
                </Grid>
            </Grid>
        </Grid>
    );
};

const Info: FC = () => {
    const global = useGlobalStyles();

    return (
        <Fade>
            <Grid container item style={{ height: '100vh' }} justify='center' id='info'>
                <Grid container item justify='center' alignItems='center' xs={8}>
                    <Grid container item alignItems='center'>
                        <Typography variant='h1'>What’s CompE+ ?</Typography>
                    </Grid>
                    <Grid container item>
                        <Typography variant='body1'>
                            CompE+ is a student led pilot program to provide students with support to find internships in the tech industry. Finding an internship within the tech industry can be
                            difficult and daunting - that’s where CompE+ comes in! CompE+ strives to provide students with services that will aid them on their internship hunt and to provide insight
                            on what it&apos;s like to work in the tech industry.
                        </Typography>
                        <br />
                        <Typography variant='body1'>
                            The CompE+ pilot program is currently under development and is slated to launch Fall 2021 for the Summer 2022 recruitment. The program will run from September - October
                            2021.
                        </Typography>
                    </Grid>
                    <Grid container item justify='center' alignItems='center' direction='column' spacing={5}>
                        <Grid container item justify='center' alignItems='center'>
                            <Typography variant='body1' style={{ fontWeight: 600 }}>
                                We are currently looking for volunteers: Resume Reviewers and Interviewers! <br />
                            </Typography>
                        </Grid>

                        <Grid container item justify='center' alignItems='center' direction='column'>
                            <Typography variant='body1' style={{ fontWeight: 600 }}>
                                If you&apos;re interested:
                            </Typography>
                            <Button size='medium' variant='contained' className={global.main_button} onClick={() => window.location.replace('/#call-to-action')}>
                                Sign up for our mailing list!
                            </Button>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </Fade>
    );
};

const Values: FC = () => {
    const classes = useStyles();

    return (
        <Fade>
            <Grid container item style={{ height: '100vh' }} justify='center' id='values'>
                <Grid container item alignItems='center' justify='center' xs={8}>
                    <Grid container item alignItems='center'>
                        <Typography variant='h1'>Values</Typography>
                    </Grid>
                    <Grid container item spacing={5}>
                        <Grid container item xs={3} className={classes.values}>
                            <img src={Practice} className={classes.svg} />
                        </Grid>
                        <Grid container item xs={9} alignItems='center'>
                            <Typography variant='h2'>
                                Practice
                                <Typography variant='body1'>
                                    Technical interviews can be difficult and requires a good amount of practice. By providing resources and services, CompE+ can help computer engineering students
                                    tackle the technical interview with greater confidence
                                </Typography>
                            </Typography>
                        </Grid>
                    </Grid>
                    <Grid container item>
                        <Grid container item xs={9} alignItems='center'>
                            <Typography align='right' variant='h2'>
                                Insight
                                <Typography align='right' variant='body1'>
                                    The tech industry can be a black box to new students. CompE+ strives to provide insight into the industry by sharing the internship experience with all computer
                                    engineering students
                                </Typography>
                            </Typography>
                        </Grid>
                        <Grid container item xs={3} className={classes.values}>
                            <img src={Industry} className={classes.svg} />
                        </Grid>
                    </Grid>
                    <Grid container item spacing={5}>
                        <Grid container item xs={3} className={classes.values}>
                            <img src={Support} className={classes.svg} />
                        </Grid>
                        <Grid container item xs={9} alignItems='center'>
                            <Typography variant='h2'>
                                Support{' '}
                                <Typography variant='body1'>
                                    Entering the tech industry can be intimidating to many, even after landing the first internship, imposter syndrome can hit hard. Fostering a positive community is
                                    an important goal of CompE+
                                </Typography>
                            </Typography>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </Fade>
    );
};

const Services: FC = () => {
    const classes = useStyles();

    return (
        <Fade>
            <Grid container item style={{ height: '100vh' }} justify='center' id='services'>
                <Grid container item alignItems='center' justify='center' xs={8}>
                    <Grid container item alignItems='center'>
                        <Typography variant='h1'>Services</Typography>
                    </Grid>
                    <Grid container item spacing={2} alignItems='center' justify='center'>
                        <Grid container item xs={12} spacing={3}>
                            <Grid container item xs={4} justify='center'>
                                <img src={ResumeReview} className={classes.services_svg} />
                            </Grid>
                            <Grid container item xs={4} justify='center'>
                                <img src={MockInterview} className={classes.services_svg} />
                            </Grid>
                            <Grid container item xs={4} justify='center'>
                                <img src={Teamwork} className={classes.services_svg} />
                            </Grid>
                        </Grid>

                        <Grid container item xs={12} spacing={3}>
                            <Grid container item xs={4} justify='center'>
                                <Typography variant='h2'>Resume Review</Typography>
                            </Grid>
                            <Grid container item xs={4} justify='center'>
                                <Typography variant='h2'>Mock Interviews</Typography>
                            </Grid>
                            <Grid container item xs={4} justify='center'>
                                <Typography variant='h2'>Community</Typography>
                            </Grid>
                        </Grid>

                        <Grid container item xs={12} spacing={3}>
                            <Grid item xs={4}>
                                <Typography align='center' variant='body1'>
                                    Gain confidence when writing resumes by getting your resume reviewed by senior CompE students to gain key insight on how to properly format and develop your resume
                                </Typography>
                            </Grid>
                            <Grid item xs={4}>
                                <Typography align='center' variant='body1'>
                                    Led by senior CompE students, practice communicating your thoughts and problem-solving skills with mock technical interviews
                                </Typography>
                            </Grid>
                            <Grid item xs={4}>
                                <Typography align='center' variant='body1'>
                                    Connect with other CompE students, gain insight on different companies that other students have worked for, and learn more about the tech industry
                                </Typography>
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </Fade>
    );
};

const CallToAction: FC = () => {
    const classes = useStyles();

    return (
        <>
            <img src={Wave2} className={classes.wave} id='call-to-action' />
            <Grid container item className={`${classes.wave_pattern} `} style={{ minHeight: '50vh' }}>
                <Fade>
                    <Grid container item justify='center' alignItems='center' xs={8} spacing={5}>
                        <Grid container item alignItems='center'>
                            <Grid item>
                                <Typography variant='h1'>Mailing List</Typography>
                            </Grid>
                            <Grid item>
                                <Typography variant='body1' style={{ fontWeight: 200 }}>
                                    <br />
                                    CompE+ is still in development 🔧 <br /> Enter your email below to sign up for our mailing list to get up-to-date information on CompE+
                                </Typography>
                            </Grid>
                        </Grid>
                        <MailChimpForm />
                    </Grid>
                </Fade>
            </Grid>
            <img src={Wave3} className={classes.wave} />
        </>
    );
};

const Footer: FC = () => {
    const classes = useStyles();

    return (
        <Grid container item className={classes.footer}>
            <Grid container item className={classes.wave_pattern} justify='center' direction='column' alignItems='center'>
                <Grid container item alignItems='center' justify='center' xs={11}>
                    <Grid container item xs={4} direction='column'>
                        <Grid item>
                            <Typography className={classes.footer_text} align='left'>
                                Contact: external@compeclub.com
                            </Typography>
                        </Grid>
                        <Grid item>
                            <Typography className={classes.footer_text} align='left'>
                                In affiliation with the{' '}
                                <Link href='https://www.compeclub.com/' color='inherit' underline='always'>
                                    CompE Club
                                </Link>
                            </Typography>
                        </Grid>
                    </Grid>
                    <Grid container item xs={4} alignItems='center' justify='center'>
                        <img src={LightGreenLogo} className={classes.footer_logo} />
                    </Grid>
                    <Grid item xs={4}>
                        <Typography className={classes.footer_text} align='right'>
                            Curious how we&apos;re building CompE+? Checkout the Github{' '}
                            <Link href='https://github.com/UAlbertaCompEClub/compe-plus' color='inherit' underline='always'>
                                repo
                            </Link>{' '}
                            😎
                        </Typography>
                    </Grid>
                </Grid>
            </Grid>
        </Grid>
    );
};

const useStyles = makeStyles((theme) => ({
    wave_pattern: {
        backgroundColor: theme.palette.primary.main,
    },
    svg: {
        width: '90%',
        height: 'auto',
    },
    wave: {
        width: '100%',
        height: 'auto',
    },
    values: {
        justifyContent: 'center',
        alignItems: 'center',
    },
    services_svg: {
        width: '80%',
        height: 'auto',
    },
    logo_1: {
        width: '50%',
        height: 'auto',
    },
    footer: {
        padding: '0',
    },
    footer_logo: {
        width: '7%',
        height: 'auto',
        margin: '1em',
    },
    text_input_root: {
        padding: '2px 4px',
        display: 'flex',
        alignItems: 'center',
        width: 400,
        borderRadius: 0,
    },
    text_input: {
        marginLeft: theme.spacing(1),
        flex: 1,
    },
    text_input_icon_button: {
        padding: 10,
    },
    text_input_divider: {
        height: 28,
        margin: 4,
    },
    footer_text: {
        color: theme.palette.primary.light,
        fontSize: 16,
        fontWeight: 400,
    },
}));

export default Landing;
