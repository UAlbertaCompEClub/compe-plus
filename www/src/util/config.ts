const AUTH0_DOMAIN = process.env.REACT_APP_AUTH0_DOMAIN || '';
const AUTH0_CLIENT_ID = process.env.REACT_APP_AUTH0_CLIENT_ID || '';
const SERVER_ENDPOINT = process.env.REACT_APP_SERVER_ENDPOINT || 'http://localhost:1337';
const SERVER_AUDIENCE = process.env.REACT_APP_SERVER_AUDIENCE || '';
const MAILCHIMP_URL = process.env.REACT_APP_MAILCHIMP_URL || '';

const AUTH0 = {
    domain: AUTH0_DOMAIN,
    clientId: AUTH0_CLIENT_ID,
};

const SERVER = {
    endpoint: SERVER_ENDPOINT,
    audience: SERVER_AUDIENCE,
};

const config = {
    auth0: AUTH0,
    server: SERVER,
    mailchimpUrl: MAILCHIMP_URL,
};

export default config;
