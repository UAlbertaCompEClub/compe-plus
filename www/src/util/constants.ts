export const COMPE_PLUS = 'CompE+';

export const ABOUT = 'About';
export const RESUME_REVIEW = 'Resume Review';
export const MOCK_INTERVIEW = 'Mock Interview';
export const COMMUNITY = 'Community';
export const COMPE_CLUB = 'CompE Club';

// Routes
export const RESUME_REVIEW_ROUTE = '/resume-review';
export const MOCK_INTERVIEW_ROUTE = '/mock-interview';
export const COMMUNITY_ROUTE = '/community';
export const REGISTER_ROUTE = '/register';
export const COMPE_CLUB_ROUTE = 'https://www.compeclub.com/';
