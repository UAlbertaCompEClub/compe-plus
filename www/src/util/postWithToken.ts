import { useAuth0 } from '@auth0/auth0-react';
import axios, { AxiosResponse } from 'axios';

export const postWithToken = async <DataType, ResponseType>(url: string, scopes: string[], data?: DataType): Promise<AxiosResponse<ResponseType> | undefined> => {
    const { getAccessTokenSilently } = useAuth0();
    try {
        const token = await getAccessTokenSilently({
            scope: scopes.join(' '),
        });
        return axios.post(url, data, {
            headers: {
                authorization: `Bearer ${token}`,
            },
        });
    } catch (e) {
        console.error(e);
    }
    return undefined;
};
