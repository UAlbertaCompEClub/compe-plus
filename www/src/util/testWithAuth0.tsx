/* eslint-disable @typescript-eslint/ban-types */
/* eslint-disable @typescript-eslint/no-explicit-any */
import { Auth0Context, Auth0ContextInterface } from '@auth0/auth0-react';
import { render, RenderResult } from '@testing-library/react';
import Enzyme, { shallow } from 'enzyme';
import React from 'react';

function shallowWithAuth0(
    component: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
    authState: Partial<Auth0ContextInterface>,
): Enzyme.ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>> {
    return shallow(component, {
        wrappingComponent: Auth0Context.Provider,
        wrappingComponentProps: {
            value: authState,
        },
    });
}

function renderWithAuth0(component: React.ReactElement<any, string | React.JSXElementConstructor<any>>, authState: Partial<Auth0ContextInterface>): RenderResult {
    return render(<Auth0Context.Provider value={authState as any}>{component}</Auth0Context.Provider>);
}

export { renderWithAuth0, shallowWithAuth0 };
