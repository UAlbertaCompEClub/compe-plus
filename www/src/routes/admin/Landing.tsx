import { useAuth0 } from '@auth0/auth0-react';
import {
    Button,
    Checkbox,
    CircularProgress,
    FormControl,
    FormControlLabel,
    FormGroup,
    Grid,
    makeStyles,
    Paper,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
    withStyles,
} from '@material-ui/core';
import React, { FC, useEffect, useState } from 'react';

import { useAdminDispatch, useAdminSelector } from '../../redux/substores/admin/adminHooks';
import getAllUserRoles from '../../redux/substores/admin/thunks/getAllUserRoles';
import useGlobalStyles from '../../styles/style';
import { UserRole } from '../../util/serverResponses';

const StyledTableCell = withStyles((theme) => ({
    head: {
        backgroundColor: theme.palette.primary.dark,
        color: theme.palette.common.white,
        fontSize: 18,
    },
}))(TableCell);

const StyledTableRow = withStyles((theme) => ({
    root: {
        '&:nth-of-type(odd)': {
            backgroundColor: theme.palette.secondary.main,
        },
    },
}))(TableRow);

interface UserTableProps {
    userRoles: userRoleTableEntries;
    action: (id: string, isUnassigned: boolean, role: string) => void;
}

const UserRoleTable: FC<UserTableProps> = (props: UserTableProps) => {
    const classes = useStyles();
    return (
        <TableContainer component={Paper}>
            <Table className={classes.table} aria-label='simple table'>
                <TableHead>
                    <StyledTableRow>
                        <StyledTableCell>Name</StyledTableCell>
                        <StyledTableCell align='right'>Roles</StyledTableCell>
                    </StyledTableRow>
                </TableHead>
                <TableBody>
                    {Object.entries(props.userRoles).map(([id, userRole]) => {
                        return (
                            <StyledTableRow key={id}>
                                <StyledTableCell component='th' scope='row'>
                                    {userRole.name}
                                </StyledTableCell>
                                <StyledTableCell align='right'>
                                    <FormControl>
                                        <FormGroup>
                                            <>
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('student')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='student'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Student'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('reviewer')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='reviewer'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Reviewer'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('interviewer')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='interviewer'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Interviewer'
                                                />
                                                <FormControlLabel
                                                    control={
                                                        <Checkbox
                                                            checked={userRole.roles.includes('admin')}
                                                            onChange={(e) => props.action(id, !e.target.checked, e.target.name)}
                                                            name='admin'
                                                            color='primary'
                                                        />
                                                    }
                                                    label='Admin'
                                                />
                                            </>
                                        </FormGroup>
                                    </FormControl>
                                </StyledTableCell>
                            </StyledTableRow>
                        );
                    })}
                </TableBody>
            </Table>
        </TableContainer>
    );
};

type userRoleTableEntries = {
    [key: string]: {
        roles: string[];
        name: string;
    };
};

type SuperSet = {
    [key: string]: number;
};

const initialUserRoleTableState: userRoleTableEntries = {};
const initialUpdatedUserRolesState: string[] = [];

// https://stackoverflow.com/questions/6229197/how-to-know-if-two-arrays-have-the-same-values/55614659#55614659
function areArraysEqualSets(a1: string[], a2: string[]) {
    const superSet: SuperSet = {};
    for (const i of a1) {
        const e = i + typeof i;
        superSet[e] = 1;
    }

    for (const i of a2) {
        const e = i + typeof i;
        if (!superSet[e]) {
            return false;
        }
        superSet[e] = 2;
    }

    for (const e in superSet) {
        if (superSet[e] === 1) {
            return false;
        }
    }

    return true;
}

const Landing: FC = () => {
    const classes = useStyles();
    const globalClasses = useGlobalStyles();
    const { userRoles, userRolesIsLoading, shouldReload } = useAdminSelector((state) => state.userRoles);
    const { getAccessTokenSilently } = useAuth0();
    const dispatch = useAdminDispatch();

    const [userRolesTable, setUserRolesTable] = useState(initialUserRoleTableState);
    const [originalUserRoles, setOriginalUserRoles] = useState(initialUserRoleTableState);
    const [updatedUserRoles, setUpdatedUserRoles] = useState(initialUpdatedUserRolesState);

    const createUserRolesTable = (userRoles: UserRole[]) => {
        return userRoles.reduce((accum, userRole) => ({ ...accum, [userRole.id]: { roles: userRole.roles, name: userRole.fullName } }), {});
    };

    const updateRole = (id: string, isUnassigned: boolean, role: string) => {
        const updatedRoles = isUnassigned ? userRolesTable[id].roles.filter((userRole) => userRole !== role) : [...userRolesTable[id].roles, role];

        if (areArraysEqualSets(originalUserRoles[id].roles, updatedRoles)) {
            setUpdatedUserRoles(updatedUserRoles.filter((userId) => userId !== id));
        } else if (!updatedUserRoles.includes(id)) {
            setUpdatedUserRoles([...updatedUserRoles, id]);
        }

        setUserRolesTable({ ...userRolesTable, [id]: { ...userRolesTable[id], roles: updatedRoles } });
    };

    useEffect(() => {
        if (shouldReload) {
            dispatch(getAllUserRoles({ tokenAcquirer: getAccessTokenSilently }));
        }
    }, [shouldReload]);

    useEffect(() => {
        if (userRoles.length > 0) {
            setUserRolesTable(createUserRolesTable(userRoles));
            setOriginalUserRoles(createUserRolesTable(userRoles));
        }
    }, [userRoles]);

    return (
        <Grid container justify='center' alignItems='center' direction='column' className={classes.pageGrid}>
            <Grid container justify='flex-start' alignItems='flex-start' direction='column'>
                <Typography variant='h1'>Users</Typography>
                <Button disabled={updatedUserRoles.length <= 0} className={globalClasses.main_button}>
                    Update Roles
                </Button>
                <Grid container justify='flex-start' alignItems='flex-start' className={classes.wrapper}>
                    {userRolesIsLoading && (
                        <Grid container justify='center' alignItems='flex-start'>
                            <CircularProgress />
                        </Grid>
                    )}
                    {!userRolesIsLoading && userRoles.length == 0 && <Typography>There are no users in CompE+, something is wrong :(</Typography>}
                    {!userRolesIsLoading && userRoles.length > 0 && <UserRoleTable userRoles={userRolesTable} action={updateRole} />}
                </Grid>
            </Grid>
        </Grid>
    );
};

const useStyles = makeStyles({
    table: {
        minWidth: 650,
    },
    wrapper: {
        marginTop: 25,
        marginBottom: 25,
    },
    pageGrid: {
        padding: 50,
    },
});

export default Landing;
