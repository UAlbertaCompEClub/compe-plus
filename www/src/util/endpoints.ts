import config from './config';

export const getMe = `${config.server.endpoint}/api/v1/users/me`;
export const getResumeReviews = `${config.server.endpoint}/api/v1/resume-reviews`;
export const getUserRole = (userId: string): string => `${config.server.endpoint}/api/v1/users/${encodeURIComponent(userId)}/roles`;

export const postUsers = `${config.server.endpoint}/api/v1/users`;

export const postResumeReviews = `${config.server.endpoint}/api/v1/resume-reviews`;
export const patchResumeReviews = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}`;
export const patchMyResumeReview = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}`;

export const postDocuments = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}/documents`;
export const getMyDocuments = (resumeReviewId: string): string => `${config.server.endpoint}/api/v1/resume-reviews/${resumeReviewId}/documents`;
