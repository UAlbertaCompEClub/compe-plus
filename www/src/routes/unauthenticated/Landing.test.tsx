import { shallow } from 'enzyme';
import React from 'react';

import { shallowWithAuth0 } from '../../util/testWithAuth0';
import Landing, { Intro } from './Landing';

describe('Landing', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Landing />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Intro', () => {
    it.each`
        isAuthenticated | friendlyName
        ${true}         | ${'when authenticated'}
        ${false}        | ${'when unauthenticated'}
    `('renders correctly', ({ isAuthenticated, friendlyName }) => {
        const wrapper = shallowWithAuth0(<Intro />, {
            isAuthenticated,
        });
        expect(wrapper).toMatchSnapshot(friendlyName);
    });
});
