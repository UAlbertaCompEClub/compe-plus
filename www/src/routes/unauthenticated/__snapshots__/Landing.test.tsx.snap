// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders correctly: when authenticated 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-wave_pattern-1"
  container={true}
  id="intro"
  item={true}
  justify="center"
  style={
    Object {
      "minHeight": "50vh",
      "paddingTop": "10vh",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    justify="center"
    spacing={5}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          maxWidth="sm"
        >
          <img
            className="makeStyles-logo_1-6"
            src="logo_black.svg"
          />
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))
          variant="h1"
        >
          CompE+
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Resume Review, Mock Interviews, Community
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={5}
    >
      <img
        className="makeStyles-svg-2"
        src="pair_programming.svg"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    item={true}
    justify="center"
    spacing={5}
    style={
      Object {
        "paddingTop": "5vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-main_button-14"
          onClick={[Function]}
          size="large"
          variant="contained"
        >
          Get Started!
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-secondary_button-15"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          Learn More
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          style={
            Object {
              "fontSize": 28,
            }
          }
          variant="h3"
        >
          Get the help you need to land your dream internship!
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Intro renders correctly: when unauthenticated 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-wave_pattern-1"
  container={true}
  id="intro"
  item={true}
  justify="center"
  style={
    Object {
      "minHeight": "50vh",
      "paddingTop": "10vh",
    }
  }
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    justify="center"
    spacing={5}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <Styled(MuiBox)
          maxWidth="sm"
        >
          <img
            className="makeStyles-logo_1-6"
            src="logo_black.svg"
          />
        </Styled(MuiBox)>
        <WithStyles(ForwardRef(Typography))
          variant="h1"
        >
          CompE+
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="h3"
        >
          Resume Review, Mock Interviews, Community
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      xs={5}
    >
      <img
        className="makeStyles-svg-2"
        src="pair_programming.svg"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
    item={true}
    justify="center"
    spacing={5}
    style={
      Object {
        "paddingTop": "5vh",
      }
    }
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
      spacing={4}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-main_button-14"
          onClick={[Function]}
          size="large"
          variant="contained"
        >
          Get Started!
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Button))
          className="makeStyles-secondary_button-15"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          Learn More
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      item={true}
      justify="center"
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          style={
            Object {
              "fontSize": 28,
            }
          }
          variant="h3"
        >
          Get the help you need to land your dream internship!
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`Landing renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  justify="center"
  style={
    Object {
      "overflowX": "hidden",
    }
  }
>
  <Intro />
  <img
    className="makeStyles-wave-3"
    src="wave_1.svg"
  />
  <Services />
  <Info />
  <CallToAction />
  <Values />
  <img
    className="makeStyles-wave-3"
    src="wave_4.svg"
  />
  <Footer />
</WithStyles(ForwardRef(Grid))>
`;
